### PR TITLE
add metrics pkg

### PR DESCRIPTION
**Pull request by ceyonur**
Adds metrics pkg and enables metrics

### CODE REVIEW

**Review by aaronbuchwald — APPROVED** [review]
LGTM - let's hold off on merging this until https://github.com/ava-labs/subnet-evm/pull/108#pullrequestreview-983791142 has been merged, and then merge this into master.

File: plugin/evm/vm.go [hunk 1]
```diff
@@ -14,16 +14,19 @@ import (
 	"sync"
 	"time"
 
+	avalanchegoMetrics "github.com/ava-labs/avalanchego/api/metrics"
```
Comment (patrick-ogrady) [inline plugin/evm/vm.go:17]: should add newline after

File: plugin/evm/vm.go [hunk 2]
```diff
@@ -14,16 +14,19 @@ import (
...
 	"github.com/ava-labs/subnet-evm/core/types"
 	"github.com/ava-labs/subnet-evm/eth/ethconfig"
-	"github.com/ava-labs/subnet-evm/metrics/prometheus"
+	"github.com/ava-labs/subnet-evm/metrics"
+	subnetEVMPrometheus "github.com/ava-labs/subnet-evm/metrics/prometheus"
 	"github.com/ava-labs/subnet-evm/node"
 	"github.com/ava-labs/subnet-evm/params"
 	"github.com/ava-labs/subnet-evm/peer"
 	"github.com/ava-labs/subnet-evm/plugin/evm/message"
+	"github.com/prometheus/client_golang/prometheus"
```
Comment (patrick-ogrady) [inline plugin/evm/vm.go:29]: should add newline before or group with `avalanchegoMetrics`

File: plugin/evm/vm.go [hunk 3]
```diff
@@ -351,6 +362,10 @@ func (vm *VM) Initialize(
 	vm.chain = ethChain
```
Comment (patrick-ogrady) [inline plugin/evm/vm.go:362]: We should port `readLastAccepted` for the `lastAcceptedBytes` block: https://github.com/ava-labs/coreth/blob/9fb7a2025a20f3a5546683571f1066cae932457c/plugin/evm/vm.go#L1708-L1732

File: plugin/evm/vm.go [hunk 4]
```diff
@@ -351,6 +362,10 @@ func (vm *VM) Initialize(
...
 	lastAccepted := vm.chain.LastAcceptedBlock()
```
Comment (patrick-ogrady) [inline plugin/evm/vm.go:363]: Let's log this out (but correctly unlike coreth): https://github.com/ava-labs/coreth-internal/blob/9fb7a2025a20f3a5546683571f1066cae932457c/plugin/evm/vm.go#L462

File: plugin/evm/vm.go [hunk 5]
```diff
@@ -375,34 +390,58 @@ func (vm *VM) Initialize(
 	vm.genesisHash = vm.chain.GetGenesisBlock().Hash()
 	log.Info(fmt.Sprintf("lastAccepted = %s", lastAccepted.Hash().Hex()))
```
Comment (patrick-ogrady) [inline plugin/evm/vm.go:391]: see earlier comment about moving this up

File: plugin/evm/vm.go [hunk 6]
```diff
@@ -351,6 +362,10 @@ func (vm *VM) Initialize(
 	vm.chain = ethChain
 	lastAccepted := vm.chain.LastAcceptedBlock()
 
+	if err := vm.initializeMetrics(); err != nil {
+		return err
+	}
+
 	// start goroutines to update the tx pool gas minimum gas price when upgrades go into effect
 	vm.handleGasPriceUpdates()
```
Comment (patrick-ogrady) [inline plugin/evm/vm.go:370]: in coreth, this happens after the peer network is created:
* https://github.com/ava-labs/coreth-internal/blob/9fb7a2025a20f3a5546683571f1066cae932457c/plugin/evm/vm.go#L473
* https://github.com/ava-labs/coreth-internal/blob/9fb7a2025a20f3a5546683571f1066cae932457c/plugin/evm/vm.go#L546

File: metrics/init_test.go [hunk 7]
```diff
@@ -0,0 +1,5 @@
+package metrics
+
+func init() {
+	Enabled = true
```
Comment (patrick-ogrady) [inline metrics/init_test.go:4]: This is not needed

File: plugin/evm/vm.go [hunk 8]
```diff
@@ -308,110 +324,115 @@ func (vm *VM) Initialize(
 		log.Warn("Chain enabled `AllowFeeRecipients`, but chain config has not specified any coinbase address. Defaulting to the blackhole address.")
 	}
 
-	// Handle offline pruning
```
Comment (patrick-ogrady) [inline plugin/evm/vm.go:311]: Just to confirm, this was just a duplicate block?

File: plugin/evm/vm.go [hunk 9]
```diff
@@ -308,110 +324,115 @@ func (vm *VM) Initialize(
...
+	vm.networkCodec, err = message.BuildCodec()
+	if err != nil {
+		return err
+	}
+
+	// initialize peer network
+	vm.Network = peer.NewNetwork(appSender, vm.networkCodec, ctx.NodeID, vm.config.MaxOutboundActiveRequests)
+	vm.client = peer.NewClient(vm.Network)
+
+	if err := vm.initializeChain(lastAcceptedHash, ethConfig); err != nil {
```
Comment (patrick-ogrady) [inline plugin/evm/vm.go:351]: nit: could put `ethConfig` on `vm` like Coreth.

**Review by patrick-ogrady — APPROVED** [review]
LGTM...minor nits but this works!

File: plugin/evm/vm.go (hunk 8 above)
Comment (ceyonur) [reply plugin/evm/vm.go:311]: yep

File: plugin/evm/vm.go (hunk 9 above)
Comment (ceyonur) [reply plugin/evm/vm.go:351]: yeah I thought it but it's not being used anywhere. so just let those being garbage collected.

**Pull request merged**